### PR TITLE
Get Summary/Calendar of Attendee Birthdays

### PR DESCRIPTION
**Pull request by migetman9**
I thought it would be nice if we were able to look at all the attendee's who have birthdays during the event, and one step higher than that being all attendee's birthdays. The CSV has been formatted to be an import for GoogleCalendar, if that suits anyone's purpose. There is no real need for this code, just some hobby scripting. This has been tested on my local box.

Thoughts and comments welcome, as well as other changes to the branch.

### CODE REVIEW

File: uber/site_sections/summary.py [hunk 1]
```diff
@@ -392,3 +392,35 @@ def volunteers_owed_refunds(self, session):
...
         }
+
+    @csv_file
+    def attendee_birthday_calendar(self, out, session, year=datetime.now(UTC).year):
+        out.writerow(['Subject', 'Start Date', 'Start Time', 'End Date', 'End Time', 'All Day Event', 'Description',
+                      'Location', 'Private'])
+        for person in session.query(Attendee).filter(Attendee.birthdate != None).all():
+            subject = "%s's Birthday" % person.full_name
+            bday = person.birthdate
+            start_date = bday.replace(year=year)
```
Comment (RobRuana) [inline uber/site_sections/summary.py]: This line will eventually blow up on you:
```python
>>> import dateparser
>>> dateparser.parse('February 29, 1996').replace(year=2017)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: day is out of range for month
```

Manipulating dates can be quite tricky. What you want to do is calculate the delta in years, and then add that delta to the birthday. This is an arduous process using the standard `datetime` library. I recommend using [dateutil](https://dateutil.readthedocs.io):
```python
>>> from dateutil.relativedelta import relativedelta
>>> birthday = dateparser.parse('February 29, 1996')
>>> delta_years = datetime.date.today().year - birthday.year
>>> birthday + relativedelta(years=delta_years)
datetime.datetime(2017, 2, 28, 0, 0)
```

Our deployment includes `dateparser`, which depends on `dateutil`, so you *can* use it without modifying our `requirements.txt`. However, if you are going to import `dateutil` directly, it's good practice to include it in `requirements.txt`, in case `dateparser` ever changes their internal dependencies:
```
python-dateutil>=2.6.0
```

Comment (RobRuana) [inline uber/site_sections/summary.py]: FYI:  in order to fix #2536, I replaced `dateparser` with `dateutil`. So now you can do:

```python
>>> from dateutil import parser as dateparser
>>> from dateutil.relativedelta import relativedelta
>>> birthday = dateparser.parse('February 29, 1996')
>>> delta_years = datetime.date.today().year - birthday.year
>>> birthday + relativedelta(years=delta_years)
datetime.datetime(2017, 2, 28, 0, 0)
```

**Comment by RobRuana** [comment]
I updated the date calculations and added unit tests. Date/time calculations are always incredibly tricky. Even when you _think_ you're doing the right thing, chances are you've gotten something wrong.

Anytime date/time calculations are involved, it's always a good idea to write some tests with a bunch of corner case dates to see where things break. Good test data will include things like:
* Leap years
* Leap days
* Future dates
* Impossible dates like Feb 29, 2017 (expecting failure)
* Date ranges that cross month boundaries
* Date ranges that cross year boundaries
* Date ranges that cross calendar eras, if applicable (BC/AD)

Never add or subtract time intervals from raw date components and expect it to work. Always use a library to add and subtract dates and intervals.

**Comment by migetman9** [comment]
Hey Rob, I've had a busy and not-so-great day/week. Seriously, thanks for cleaning this up and leaving the notes you have. Thank you.
